lib/restbase.js

```diff
@@ -168,6 +172,10 @@ RESTBase.prototype.defaultWebRequestHandler = function(req) {
     // Enforce the usage of UA
     req.headers = req.headers || {};
     req.headers['User-Agent'] = req.headers['User-Agent'] || this.rb_config.user_agent;
+    if (this._accessRestrictions.length > 0
+        && req.headers && req.headers['x-internal-request']) {
```

Comment:
This header can be injected in the request. We'd either have to set it to false unconditionally when receiving a request, or use some other mechanism to distinguish internal from external services.

I'm wondering if a whitelist (regexp?) to identify internal services would be cleaner than the x-internal-request bit here. For WMF-internal services, this could be something like `/^http:\/\/[^/]+\.wmnet(:[0-9]+)?\//`.